src/gui/pageposition/PagePositionHandler.cpp

```diff
@@ -129,38 +94,34 @@ XojPageView* PagePositionHandler::getViewAt(int x, int y, PagePositionCache* cac
 		return NULL;
 	}
 
-	return pp->getViewAt(x, y);
+	return pp->pv;
 }
 
-PagePosition* PagePositionHandler::binarySearch(PagePosition** sortedArray, int first, int last, int y, int& index)
+PagePosition* PagePositionHandler::linearSearch( int x,int y, int& index)
```

Comment:
You should use a better search algorithm for really big documents. But that is not really a blocker and can be merged later on.

Reply:
I agree.
Note that the cache system is still there so it will try the last page found first.
The layoutMapper already knows enough about the layout to speed up any search so it might be a candidate to take on that function too.